src/hotspot/share/gc/g1/g1_globals.hpp

```diff
@@ -73,7 +73,11 @@
                                                                             \
   product(bool, G1EvacuationFailureALotDuringMixedGC, true,                 \
           "Force use of evacuation failure handling during mixed "          \
-          "evacuation pauses")
+          "evacuation pauses")                                              \
+                                                                            \
+  product(uintx, G1EvacuationFailureALotCSetPercent, 100,                   \
```

Comment:
Why `uintx`? Does `uint` work?

Reply:
I will refine the explanation specifically.
As Thomas suggested, an random selection will be added later, I plan to add another bool option to indicate if it's random selection or not, so will keep `G1EvacuationFailureALotCSetPercent`.

Reply:
Another, imo better way would be allowing a string for the type of selection algorithm, like `VerifyGCType`, with current possible values being `NthObject` (or just default) and `FirstRegions` or so.

Reply:
That could be extended to `RandomRegions` later.

The name of the flag should change then though, maybe something like `G1EvacuationFailureALotSelectionType` (random name right now, probably you will come up with something better).

Reply:
I think you mean we add this option when implement https://bugs.openjdk.java.net/browse/JDK-8277851, am I right? If possible, sure, it make sense to me, it's much easier to extend the algorithms in the future.